include/ginkgo/core/base/exception_helpers.hpp

```diff
@@ -117,6 +117,18 @@ inline dim<2> get_size(const dim<2> &size) { return size; }
 }  // namespace detail
 
 
+/**
+ *Asserts that _val1 and _val2 are equal.
+ *
+ *@throw BadDimension if _val1 is different from _val2.
+ */
+#define GKO_ASSERT_EQ(_val1, _val2)                                        \
```

Comment:
`GKO_ASSERT_EQ` throwing `BadDimension` is confusing.
I would change the name like `GKO_ASSERT_PERM_LEGAL` or something else.